server/shared/src/main/scala/org/http4s/server/middleware/RequestId.scala

```diff
@@ -44,65 +42,61 @@ object RequestId {
 
   val requestIdAttrKey: Key[String] = Key.newKey[SyncIO, String].unsafeRunSync()
 
-  def apply[G[_], F[_]](http: Http[G, F])(implicit G: Sync[G]): Http[G, F] =
+  def apply[G[_]: Monad: UUIDGen, F[_]](http: Http[G, F]): Http[G, F] =
```

Comment:
I removed all the deprecations from `RequestId` (those coming from the current blob), as it’s in the spirit of 1.0, and will likely only require clients to recompile their code